pkg/controller/common/association/gc_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package association
+
+import (
+	"testing"
+
+	apmtype "github.com/elastic/cloud-on-k8s/pkg/apis/apm/v1beta1"
+	kibanatype "github.com/elastic/cloud-on-k8s/pkg/apis/kibana/v1beta1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/user"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	ApmAssociationLabelName         = "apmassociation.k8s.elastic.co/name"
+	ApmAssociationLabelNamespace    = "apmassociation.k8s.elastic.co/namespace"
+	KibanaAssociationLabelName      = "kibanaassociation.k8s.elastic.co/name"
+	KibanaAssociationLabelNamespace = "kibanaassociation.k8s.elastic.co/namespace"
```

Comment:
can't we grab the labels from where they are already defined?

Reply:
It would introduce a cycle: controller/common/association -> controller/apmserverelasticsearchassociation -> controller/common/association

The value of those variables is not important in the test, I just used the real ones here to help people understanding the test code.

Reply:
got it 👍